launcher-aiming-control

cjdrake42 opened the pull request.

cjdrake42 committed:
Add files via upload

cjdrake42 committed:
Update launcher-aiming-control.md

cjdrake42 committed:
Update launcher-aiming-control.md

fixed equations

cjdrake42 committed:
Update launcher-aiming-control.md

Added note for bill of materials

JesseTNRoberts commented on Documentation/Signoffs/launcher-aiming-control.md:
```diff
@@ -0,0 +1,84 @@
+## Functionality
+The launch aiming control unit handles the different functions within the launcher. This includes controlling the signals for the motors for the launcher aiming subsystem, sending a signal for the buzzer to indicate that the launcher is about to fire, and sending a signal to the launcher firing subsystem indicating to fire.
+## Constraints
+C1: The launch aiming control unit shall have the sufficient amount of pins to control the each subsystem
```
How many is that?

JesseTNRoberts reviewed (commented):
The constraints for this system have not been thought through. There's no way to design the system unless you have a clear understanding of what it must do.

cjdrake42 committed:
Update launcher-aiming-control.md

cjdrake42 committed:
Add files via upload

cjdrake42 committed:
Update launcher-aiming-control.md

cjdrake42 committed:
Update launcher-aiming-control.md

cjdrake42 committed:
Update launcher-aiming-control.md

JesseTNRoberts reviewed (commented):
This looks like a signoff for the raspberry pi and controls to the launcher. But it's a signoff for the control of the aiming? 

This is a good signoff and I would be willing to approve but I'm confused at what I'm agreeing to.

cjdrake42 committed:
Update and rename launcher-aiming-control.md to launcher-control.md

Clarified what signoff is by changing terminology of the system. The signoff name is the launcher control system rather than the launcher aiming because it controls the launcher actuators based on sensor inputs

cjdrake42 committed:
Update launcher-control.md

cjdrake42 commented:
I believe that initially in our conceptual design the launcher aiming system's responsibility was to take in sensor inputs, move the launcher to position, calculate time to fire for the firing system, and send the signal to the firing system. When approaching the detailed design, it was determined that it would make more sense to split the subsystem's parts into two signoffs: the launcher aiming system which highlights the motors needed to move the launcher and the launcher aiming control which provides the control for all of the tasks in the launcher aiming system. I agree that the terminology for the launcher aiming control is not clear because it is performing tasks outside the scope of the launcher aiming system. As a result I have changed the terminology of the system from launcher aiming control to launcher control in the signoff markdown file.

micahr234 commented:
Your block diagram shows that the time to fire is calculated AFTER the launcher is moved to the detection path. But the t_fire (time to fire) variable in your equations is dependent on the t_aim (time to aim). Seems like you will have already aimed when you calculate t_fire. Can you explain?

cjdrake42 committed:
Update launcher-control.md

cjdrake42 commented:
Added a clarification for what t_aim and t_fire is and how they interact with each other
